_posts/2019-03-06-generating-print-quality-plots.md

```diff
@@ -0,0 +1,128 @@
+Adding the `res` parameter fixes up the resolution, but it also rescales the axes and text. Therefore, you need to increase the height and the width.
+
+```R
+png('heatmap3.png', pointsize=10, width=1400, height=960, res=300)
+filled.contour(lons,lats,area,
+               xlim = range(lons), ylim = range(lats),
+               plot.axes = {axis(1);axis(2);
+                 map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+![Doubled width and height]({{ "/images/heatmap3.PNG" | absolute_url }})
+
+That's better.
+
+What if we want an even higher resolution, say 600dpi. So we change the first line to:
+```R
+png('C:/Users/Danny/Documents/heatmap4.png', pointsize=10, width=2800, height=2000, res=600)
+```
+and run the plot again.
+This time you might get the following error: `Error in plot.new() : figure region too large`.
+
+What this means is that your figure doesn't fit in the default device screen that R has set. What does that mean? Well, when you run a `png()` command, you are actually telling R to set up a PNG output device on which to display your plot. That device has certain default parameters, including dimensions and margin sizes. In this case, the plot you have specified doesn't fit in the device's spatial parameters.
+
+You may be able to solve this problem by changing the device's margin sizes. This may require a bit of fiddling to get them right. Margin sizes are measured in units of lines of text. (A plot has a specific text size.)
+
+By adding a `par(mar=...)` command, my error went away.
+```R
+png('C:/Users/Danny/Documents/heatmap5.png', pointsize=10, width=2800, height=2000, res=600)
+par(mar=c(5, 4, 4, 2))
+filled.contour(lons,lats,area,
+               xlim = range(lons), ylim = range(lats),
+               plot.axes = {axis(1);axis(2);
+                 map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+
+You may also get an error message saying that the margins are too big. That might indicate that:
+1. You have to make them smaller, or
+2. You haven't set your plot dimensions to be sufficiently large, so there isn't enough space to fit the margins.
+
+## Changing the Colour Palette
+
+R's default choice of colours is not to everyone's taste and those submitting theses have been known to be asked to change the colour palette.
+
+To set the colour palette for a plot, you just have to create a vector of colour names.
+```R
+# create a vector with 24 shades ranging between the colours mentioned
+colours = colorRampPalette(c("blue", "white", "pink", "red"))(24)
+png('heatmap4.png', pointsize=10, width=1400, height=960, res=300, units="px")filled.contour(lons,lats,area, col=colours,
+               xlim = range(lons), ylim = range(lats),
+               plot.axes = {axis(1);axis(2);
+                 map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+![Fixed up the palette]({{ "/images/heatmap4.PNG" | absolute_url }})
+
```

Comment:
Might be worth mentioning how the number of colours in the palette lines up with the levels in the contour plot. Not  enough levels in the palette and the colour will cycle, too few and the top colors aren't represented.